website/docs/r/compute_instance.html.markdown

```diff
@@ -54,6 +54,89 @@ resource "google_compute_instance" "default" {
 }
 ```
 
+## Best practice: service account, scopes, and IAM roles
+The
+[Service accounts - Access scopes](https://cloud.google.com/compute/docs/access/service-accounts#accesscopesiam)
+documentation explains that access scopes are the legacy
+method of specifying permissions for an instance. Current
+[best practice](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#best_practices)
+is to assign the instance a service account, which is
+granted IAM roles for only the resources that it needs.
+
+You cannot set only IAM roles on the service account and omit
+access scopes when creating an instance. The level of access a
+service account has is determined by a combination of access
+scopes and IAM roles so you must configure both access scopes
+and IAM roles for the service account to work properly. For
+more details, see the
+[Service accounts - Permissions](https://cloud.google.com/compute/docs/access/service-accounts#service_account_permissions)
+docs.
+
+When following the best practice of utilizing IAM Roles,
+you should specify the `cloud-platform` scope to allow
+full access to all Google Cloud APIs, so that the IAM
+permissions of the instance are completely determined
+by the IAM roles of the service account:
+
+```hcl
+resource "google_service_account" "web_server" {
+  // This service account will be assigned to
+  // web server instances and will be granted
+  // IAM roles that define the minimum set of
+  // required permissions.
+  account_id = "web-server"
```

Comment:
I already encounter [soft delete](https://cloud.google.com/iam/docs/creating-managing-service-accounts#undeleting_a_service_account) in GCP with service when playing with `terraform apply/destroy`

Since this time I use random provider to get new name on `apply/destroy`